app/src/main/java/com/hapramp/ui/activity/TransferActivity.java

```diff
@@ -0,0 +1,182 @@
+  TextView currencySelectorSteem;
+  @BindView(R.id.currency_selector_container)
+  RelativeLayout currencySelectorContainer;
+  private int currentCurrencyMode = 0;
+  private double mSteemBalance = 0;
+  private double mSBDBalance = 0;
+  private String finalTransferAmount = "";
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_transfer);
+    ButterKnife.bind(this);
+    collectExtra();
+    attachListeners();
+    backBtn.setTypeface(FontManager.getInstance().getTypeFace(FontManager.FONT_MATERIAL));
+  }
+
+  private void collectExtra() {
+    Intent intent = getIntent();
+    if (intent != null) {
+      mSBDBalance = Double.parseDouble(intent.getExtras().getString(EXTRA_SBD_BALANCE, "0 SBD")
+        .split(" ")[0]);
+      mSteemBalance = Double.parseDouble(intent.getExtras().getString(EXTRA_STEEM_BALANCE, "0 STEEM")
+        .split(" ")[0]);
+      setCurrencyMode(Cmode_sbd);
+    }
+  }
+
+  private void attachListeners() {
+    backBtn.setOnClickListener(new View.OnClickListener() {
+      @Override
+      public void onClick(View view) {
+        finish();
+      }
+    });
+    currencySelectorSbd.setOnClickListener(new View.OnClickListener() {
+      @Override
+      public void onClick(View view) {
+        setCurrencyMode(Cmode_sbd);
+      }
+    });
+
+    currencySelectorSteem.setOnClickListener(new View.OnClickListener() {
+      @Override
+      public void onClick(View view) {
+        setCurrencyMode(Cmode_steem);
+      }
+    });
+    continueBtn.setOnClickListener(new View.OnClickListener() {
+      @Override
+      public void onClick(View view) {
+        if (validateAmount()) {
+          openTransactionPage();
+        }
+      }
+    });
+    cancelBtn.setOnClickListener(new View.OnClickListener() {
+      @Override
+      public void onClick(View view) {
+        finish();
+      }
+    });
+  }
+
+  private void setCurrencyMode(int mode) {
+    currentCurrencyMode = mode;
+    if (mode == Cmode_sbd) {
+      currencySelectorSbd.setBackgroundColor(Color.parseColor("#3F72AF"));
+      currencySelectorSteem.setBackgroundColor(Color.parseColor("#1f000000"));
+      currencySelectorSbd.setTextColor(Color.parseColor("#ffffff"));
+      currencySelectorSteem.setTextColor(Color.parseColor("#8a000000"));
+      balanceTv.setText(String.format("Your Balance: %s SBD", mSBDBalance));
+    } else {
+      currencySelectorSteem.setBackgroundColor(Color.parseColor("#3F72AF"));
+      currencySelectorSbd.setBackgroundColor(Color.parseColor("#1f000000"));
+      currencySelectorSteem.setTextColor(Color.parseColor("#ffffff"));
+      currencySelectorSbd.setTextColor(Color.parseColor("#8a000000"));
+      balanceTv.setText(String.format("Your Balance: %s STEEM", mSteemBalance));
+    }
+  }
+
+  private boolean validateAmount() {
+    if (usernameEt.getText().toString().trim().length() == 0) {
+      toast("Enter username");
```

Comment:
`Username missing` would be a better IMO

Reply:
Ok!